docs/addons/vpc-cni.md

```diff
@@ -8,43 +8,124 @@ Installing VPC CNI as [Amazon EKS add-on](https://docs.aws.amazon.com/eks/latest
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.VpcCniAddOn();
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+
+Pattern # 2 : Custom networking with new Secondary CIDR ranges. This pattern will first create Secondary CIDRs and Secondary Subnets with specified range of CIDRs as shown below in `resourceProvider` command. Then the VPC CNI addon will setup custom networking based on the parameters `awsVpcK8sCniCustomNetworkCfg`, `eniConfigLabelDef: "topology.kubernetes.io/zone"` for your Amazon EKS cluster workloads with created secondary subnet ranges to solve IP exhaustion.
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.VpcCniAddOn({
+  customNetworkingConfig: {
+      subnets: [
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet0"),
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet1"),
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet2"),
+      ]   
+  },
+  awsVpcK8sCniCustomNetworkCfg: true,
+  eniConfigLabelDef: 'topology.kubernetes.io/zone'
+});
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .resourceProvider(blueprints.GlobalResources.Vpc, new VpcProvider(undefined,"10.64.0.0/24",["10.64.0.0/25","10.64.0.128/26","10.64.0.192/26"],))
+  .build(app, 'my-stack-name');
+```
+
+Pattern # 3 : Custom networking with custom VPC and Secondary Subnets. This pattern will use the custom VPC ID and Secondary subnet IDs passed by the user to create the blueprints stack. Then the VPC CNI addon will setup custom networking based on the parameters `awsVpcK8sCniCustomNetworkCfg`, `eniConfigLabelDef: "topology.kubernetes.io/zone"` for your Amazon EKS cluster workloads with passed secondary subnet ranges to solve IP exhaustion. 
+
+Note : 
+When you are passing your own Secondary subnets using this pattern, Please make sure the tag `Key: kubernetes.io/role/internal-elb", Value: "1"` is added to your secondary subnets. Please check out [Custom Networking Tutorial](https://docs.aws.amazon.com/eks/latest/userguide/cni-custom-network.html) to learn how custome networking is manually setup on your Amazon EKS cluster.
+
 ```typescript
 import 'source-map-support/register';
 import * as cdk from 'aws-cdk-lib';
 import * as blueprints from '@aws-quickstart/eks-blueprints';
 
 const app = new cdk.App();
 
-const addOn = new blueprints.addons.VpcCniAddOn('v1.7.5-eksbuild.2');
+const addOn = new blueprints.addons.VpcCniAddOn({
+  customNetworkingConfig: {
+      subnets: [
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet0"),
```

Comment:
Oh, I see you are addressing my previous comment, however, the name `blueprint-construct` should be dropped from this as it is only relevant for our e2e test.

Reply:
Done,